server/plugin.go

```diff
@@ -119,7 +119,7 @@ func (p *Plugin) OnActivate() error {
 func (p *Plugin) getOAuthConfig() *oauth2.Config {
 	config := p.getConfiguration()
 
-	scopes := []string{"api", "read_user"}
+	scopes := []string{}
```

Comment:
@maisnamrajusingh Why are these scopes no longer needed?

Reply:
because when the scopes in the code are filled with "api and read_user" the function is still useless if the user doesn't choose any of the scopes options on gitlab.com
so I'm looking for the safest condition, which is to make the scopes very minimal or just leave it blank

-- My reasons are below

Previously, I tried conditions with scopes filled with 'api' only or empty, then scopes on oauth gitlab.com checked only 'api'.
When doing '/gitlab connect', mattermost and gitlab.com can connect

whereas, when tested with scopes on code filled (api & read_user), and scopes on oauth gitlab.com checked only api.
when doing '/gitlab connect', there is an error 'request scopes is invalid'. 

so, my best advice is to minimize the scope of the code.

Reply:
@mickmister

Reply:
Okay, so it seems we aren't supposed to provide scopes anymore? Similar to what Zoom did with their OAuth apps. I think you instead specify the scopes in GitLab's UI, so they make it so you can't do it via the OAuth connect process anymore.